### PR TITLE
[covid vaccine] Add a masked ssn widget

### PR DESCRIPTION
**Pull request by ncksllvn**
## Description
Add masking to the first five digits of the ssn when the field loses focus.  Unmask when the field is focused. 

related to: https://github.com/department-of-veterans-affairs/va.gov-team/issues/17112

## Testing done
manual testing

## Screenshots
Without dashes:
image
image

With Dashes:
image
image

Bad data entered
image

## Acceptance criteria
- [x] First five digits of SSN are masked when the field loses focus
- [x] SSN is unmasked when the field is focused. 
- [x] SSN field validation still works as expected 
- [ ] Field can still be read by screen readers when focused 

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/coronavirus-vaccination/components/MaskedSSNWidget.jsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import React, { useState, useCallback } from 'react';
+import TextWidget from 'platform/forms-system/src/js/widgets/TextWidget';
+
+export default function MaskedSSNWidget({ value, onChange, ...props }) {
+  const [visibleValue, setVisibleValue] = useState(value);
+
+  const onFocus = useCallback(
+    () => {
+      setVisibleValue(value);
+      console.log(value);
```
Comment (ncksllvn) [inline src/applications/coronavirus-vaccination/components/MaskedSSNWidget.jsx:10]: Forgot to remove this

**Review by 1Copenut — APPROVED** [review]
LGTM! I'll review for NVDA and JAWS in staging after the branch is merged.

**Review by stephenBarrs — APPROVED** [review]

File: src/applications/coronavirus-vaccination/components/MaskedSSNWidget.jsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import React, { useState, useCallback } from 'react';
```
Comment (billfienberg) [inline src/applications/coronavirus-vaccination/components/MaskedSSNWidget.jsx:1]: Leaving the `useCallback` docs here as a breadcrumb:
https://reactjs.org/docs/hooks-reference.html#usecallback

**Review by billfienberg — APPROVED** [review]

**Review by batemapf — APPROVED** [review]

File: src/platform/forms-system/src/js/widgets/TextWidget.jsx [hunk 3]
```diff
@@ -23,6 +23,7 @@ export default function TextWidget(props) {
     onBlur: () => props.onBlur(props.id),
     onChange: event =>
       props.onChange(event.target.value ? event.target.value : undefined),
+    onFocus: props.onFocus,
```
Comment (cvalarida) [inline src/platform/forms-system/src/js/widgets/TextWidget.jsx:26]: Bonus points for adding it to `propTypes`, but I see that like none of these props are there, so I guess kinda whatever.

**Review by cvalarida — APPROVED** [review]
Design system change: Adding an `onFocus` prop to `TextWidget`.

:shipit:

**Pull request merged**